### ISSUE TITLE
Question: Is queryf silently correcting mistakes considered a bug?

### ISSUE THREAD

**Issue by lexidor** [slackhq/hack-sql-fake#71]
# Issue template for hack-sql-fake

### Please select the options that apply

 - [ ] Feature request
 - [x] Question
 - [ ] Suggestion
 - [ ] Typechecker errors
 - [ ] Incorrect result
 - [ ] Cripplingly bad performance
 - [ ] Crash
 
#### Question

Would it be considered a bug if the [queryf implementation](https://github.com/slackhq/hack-sql-fake/blob/main/src/AsyncMysql/QueryFormatter.php) of SQL Fake silently "corrects" errors which cause the real queryf to throw? A lot of strict validation is missing from the SQL Fake implementation and I would love to make it as strict as the real queryf. For example `SELECT "hi"` throws in the real queryf.

I would live to submit a PR aligning queryf with queryf. This would break BC left, right, and center. But I believe this would be for the better. Would this be appreciated?

**Comment by ssandler** [comment]
Yes, I would like the library to be as strict as Hack's implementation, would definitely welcome PRs fixing this!

**Comment by lexidor** [comment]
Alright, I'll submit a PR that enforces Hack's rules.
